packages/runtime/container-runtime/src/test/getPendingBlobs.spec.ts

```diff
@@ -112,4 +112,39 @@ describe("getPendingLocalState", () => {
 		assert.strictEqual(summaryData2.ids.length, 2);
 		assert.strictEqual(summaryData2.redirectTable.size, 2);
 	});
+
+	it("upload blob while getting pending state", async () => {
+		await runtime.attach();
+		await runtime.connect();
+		const blob = IsoBuffer.from("blob", "utf8");
+		const handleP = runtime.createBlob(blob);
+		await runtime.processBlobs();
+		const blob2 = IsoBuffer.from("blob2", "utf8");
+		const handleP2 = runtime.createBlob(blob2);
+		const pendingStateP = runtime.getPendingLocalState(true);
```

Comment:
what about when waitForBlobs is false?

Reply:
I'll add tests for it

Reply:
https://dev.azure.com/fluidframework/internal/_workitems/edit/5391 item to track this